src/qt/guiutil.cpp

```diff
@@ -243,6 +243,7 @@ QString HtmlEscape(const QString& str, bool fMultiLine)
     if(fMultiLine)
     {
         escaped = escaped.replace("\n", "<br>\n");
+        escaped = escaped.replace(" ", "&nbsp;");
```

Comment:
I think it's slightly unexpected for this to be part of the responsibility of a function called `HtmlEscape`. Especially to do this only on fMultiLine - there's nothing inherently multiline about replacing spaces.
If this is still necessary with monospace - maybe add a separate flag for this step?